java/ql/automodel/publish.sh

```diff
@@ -112,18 +122,39 @@ popd
 # and add a new file
 #  ./src/change-notes/released/<version>.md
 
-if [ -z "${1:-}" ]; then
-  # If we used the latest release of codeml-automodel, then we need to return to the current branch
-  git checkout "$CURRENT_BRANCH"
-fi
+# Get the filename of the most recently created file in ./src/change-notes/released/*.md
+# This will be the file for the new release
+NEW_CHANGE_NOTES_FILE=$(ls -t ./src/change-notes/released/*.md | head -n 1)
+
+mv ./src/CHANGELOG.md ./src/CHANGELOG.md.dry-run
+mv ./src/codeql-pack.release.yml ./src/codeql-pack.release.yml.dry-run
+mv ./src/qlpack.yml ./src/qlpack.yml.dry-run
+mv "$NEW_CHANGE_NOTES_FILE" ./src/change-notes/released.md.dry-run
 
-# Add the updated files to the current branch
-git add ./src/CHANGELOG.md
-git add ./src/codeql-pack.release.yml
-git add ./src/qlpack.yml
-git add ./src/change-notes/released/*
-echo "Added the following updated version files to the current branch:"
-git status -s
+# If --override-release was not specified, then we need to checkout the original branch
+if [ "$1" != "--override-release" ] && [ "$2" != "--override-release" ]; then
+  echo "Checking out the original branch"
+  git checkout "$CURRENT_BRANCH" --force
+fi
 
-echo "Automodel packs successfully published. Local files have been modified. Please commit and push the version changes and then merge into main."
+if [ "$1" = "--dry-run" ] || [ "$2" = "--dry-run" ]; then
+  echo "Inspect the updated dry-run version files:"
+  ls -l ./src/*.dry-run
+  ls -l ./src/change-notes/*.dry-run
+else
```

Comment:
The CLI has some built in change note processing. It's not very well documented since it's only used internally. See the `codeql pack release` and `codeql pack post-release` commands. This is how we manage the versioning of the codeql language packs. I think your script here replicates a lot of that work. 

However, to use this feature will require changing the folder layout of your change notes a bit. So, I don't think it is a blocker to not be using this command, but would be something good to move to in the future.

Reply:
I am using those commands. The extra complication here is that, in the standard use-case, I want to publish a specific `sha` of the `automodel` queries, which requires switching to a detached HEAD state, and then switching back to the developer's branch while retaining the updated version files (for inspection and then merging to `codeql` `main`).